contracts/YakVault.sol

```diff
@@ -177,41 +150,76 @@ contract YakVault is YakERC20, Ownable {
      */
     function removeStrategy(address strategy) public onlyOwner {
         require(strategy != activeStrategy, "YakVault::removeStrategy, cannot remove activeStrategy");
-        require(strategy != address(depositToken), "YakVault::removeStrategy, cannot remove deposit token");
-        require(supportedDepositTokens[strategy] == true, "YakVault::removeStrategy, not supported");
-        _revokeApproval(address(depositToken), strategy);
-        supportedDepositTokens[strategy] = false;
-        for (uint i = 0; i < supportedStrategies.length; i++) {
-            if (strategy == supportedStrategies[i]) {
-                supportedStrategies[i] = supportedStrategies[supportedStrategies.length - 1];
-                supportedStrategies.pop();
-                break;
-            }
-        }
+        require(supportedStrategies.contains(strategy) == true, "YakVault::removeStrategy, not supported");
+        require(getDeployedBalance(strategy) == 0, "YakVault::cannot remove strategy with funds");
```

Comment:
It is possible that a strategy has a deployed balance that is not withdrawable. We should be defensive in these scenarios, including in the withdraw loop